eventuals/catch.h

```diff
@@ -178,10 +182,76 @@ struct _Catch final {
 
   template <typename Value_, bool has_all_, typename... CatchHandlers_>
   struct Builder final {
+    template <typename... T>
+    struct CatchErrorsUnpacker {
+      using Errors = std::tuple<>;
+    };
+
+    // Helper that returns a 'std::tuple' with all errors,
+    // that are handled by current 'Catch'.
+    template <typename CatchHandler, typename... Tail>
+    struct CatchErrorsUnpacker<CatchHandler, Tail...> {
+      using Errors = tuple_types_union_t<
+          std::tuple<typename CatchHandler::Error>,
+          typename CatchErrorsUnpacker<Tail...>::Errors>;
+    };
+
+    template <typename CatchHandler>
+    struct CatchErrorsUnpacker<CatchHandler> {
+      using Errors = std::tuple<typename CatchHandler::Error>;
+    };
+
+    // Helper that creates the right tuple of 'Errors' for 'Catch'.
+    // If 'Catch' handles some type from upstream 'Errors' it won't be
+    // added to the result errors tuple.
+    template <
+        typename CatchErrors,
+        typename UpstreamErrors,
+        typename Processed = std::tuple<>,
+        size_t I = 0,
+        bool End = I == std::tuple_size_v<UpstreamErrors>>
+    struct Intersection {
```

Comment:
This isn't "intersection", it's set subtraction!

Let's move this to be a helper in `type-traits.h` and call it `tuple_types_subtract` and let's have it look like: `tuple_types_subtract_t<Errors, CaughtErrors>`. See [here](https://github.com/3rdparty/eventuals/blob/main/eventuals/type-traits.h#L171) for an example.

Also, let's implement this by using `tuple_types_contains`, that is, create a "new" set that only includes things from the left hand side `Errors` if the type is _not_ `tuple_types_contains` in the right hand side.

Reply:
Actually it uses `tuple_types_contains` at `std::disjunction` now.